tests/test_server.py

```diff
@@ -75,18 +75,20 @@ def test_health_check(test_client: TestClient) -> None:
     assert response.json() == {"status": "healthy"}
 
 
-@patch("codegate.api.v1_processing.fetch_latest_version", return_value="foo")
-def test_version_endpoint(mock_fetch_latest_version, test_client: TestClient) -> None:
+@patch("codegate.api.v1._get_latest_version")
```

Comment:
Ideally this test would use a patched out UpdateClient instance, but it's difficult to patch out a singleton with the mocks. 

This sort of testing would be much easier if the V1 routes were defined in a class, and all dependencies were supplied via the constructor.